example/lib/star_wars/planet_list.dart

```diff
@@ -0,0 +1,150 @@
+          Provider.value(value: api),
+          Provider.value(value: store),
+          Provider.value(value: store.state),
+        ],
+        child: HookBuilder(
+          builder: (context) {
+            final state = useAppState();
+            final store = useAppStore();
+            final isLoadingState = useState(false);
+
+            final fetchAndDispatchPlanets =
+                useMemoized<FetchAndDispatchPlanets>(
+              () => ([String url]) async {
+                isLoadingState.value = true;
+                final page = await api.getPlanets(url);
+                store.dispatch(SetPlanetPageAction(page));
+                isLoadingState.value = false;
+              },
+              [store],
+            );
+
+            final buttonAlignment = useMemoized(
+              () {
+                if (null == state.planetPage.previous) {
+                  return MainAxisAlignment.end;
+                }
+                if (null == state.planetPage.next) {
+                  return MainAxisAlignment.start;
+                }
+                return MainAxisAlignment.spaceBetween;
+              },
+              [state],
+            );
+
+            /// load the first planet page but only on the first build
+            useEffect(
+              () {
+                fetchAndDispatchPlanets(null);
+                return () {};
+              },
+              const [],
+            );
+
+            return Provider.value(
+              value: fetchAndDispatchPlanets,
+              child: CustomScrollView(
+                slivers: <Widget>[
+                  if (isLoadingState.value)
+                    SliverToBoxAdapter(
+                      child: Center(child: const CircularProgressIndicator()),
+                    ),
+                  if (!isLoadingState.value)
+                    SliverToBoxAdapter(
+                      child: Row(
+                        mainAxisAlignment: buttonAlignment,
+                        children: <Widget>[
+                          if (null != state.planetPage.previous)
+                            _LoadPageButton(
+                              next: false,
+                            ),
+                          if (null != state.planetPage.next)
+                            _LoadPageButton(
+                              next: true,
+                            )
+                        ],
+                      ),
+                    ),
+                  if (!isLoadingState.value &&
+                      state.planetPage.results.isNotEmpty)
+                    _List(),
+                ],
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+class _LoadPageButton extends HookWidget {
+  final bool next;
+
+  _LoadPageButton({this.next = true}) : assert(next != null);
+
+  @override
+  Widget build(BuildContext context) {
+    final state = useAppState();
+    final fetchAndDispatch = useFetchAndDispatchPlanets();
```

Comment:
Ah, so _that's_ the use for the Provider of the function `fetchAndDispatch`

It's a pretty bad idea to provider low level functions directly like that.
Wrap these functions in a class instead :)

Reply:
provided a function cause i exactly knew it only would be this single function :)

Reply:
Providing primitive objects like int or Functions is anti-pattern
If another store wants to expose a function with the same prototype, it'd break this one